compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -406,6 +456,106 @@ void OMR::CodeGenerator::lowerTrees()
       }
 
    self()->postLowerTrees();
+
+   if (comp()->getOption(TR_SplitWarmAndColdBlocks) &&
```

Comment:
As this will add nearly 100 lines of code applicable only when warm/cold block splitting is enabled, I think it should be moved into its own function for readability.

Reply:
Right, I think I will move all the new code in this file into a new function which will have an extra pass through the trees and then some trees modification.

Reply:
Done.